misc/mdns/src/behaviour.rs

```diff
@@ -37,28 +37,46 @@ use wasm_timer::{Delay, Instant};
 /// them to the topology.
 pub struct Mdns<TSubstream> {
     /// The inner service.
-    service: MdnsService,
+    service: MaybeBusyMdnsService,
 
     /// List of nodes that we have discovered, the address, and when their TTL expires.
     ///
     /// Each combination of `PeerId` and `Multiaddr` can only appear once, but the same `PeerId`
     /// can appear multiple times.
+    //
+    // TODO: Why this optimization? Is mdns ever within the hot path of an
```

Comment:
This isn't an optimization. It is necessary in order to store the `Multiaddresses` of the nodes we know about. Without it, you can't implement `addresses_of_peer`.

Reply:
Why can this not be a simple heap allocated `Vec` instead of a *maybe* stack allocated `SmallVec`?

Reply:
I guess it doesn't really matter.